lerobot/common/robot_devices/cameras/opencv.py

```diff
@@ -0,0 +1,285 @@
+# Especially useful during data collection when other threads are used
+# to save the images.
+cv2.setNumThreads(1)
+import numpy as np
+
+from lerobot.common.robot_devices.cameras.utils import save_color_image
+from lerobot.common.utils.utils import capture_timestamp_utc
+
+
+def find_camera_indices(raise_when_empty=False, max_index_search_range=60):
+    camera_ids = []
+    for camera_idx in range(max_index_search_range):
+        camera = cv2.VideoCapture(camera_idx)
+        is_open = camera.isOpened()
+        camera.release()
+
+        if is_open:
+            print(f"Camera found at index {camera_idx}")
+            camera_ids.append(camera_idx)
+
+    if raise_when_empty and len(camera_ids) == 0:
+        raise OSError(
+            "Not a single camera was detected. Try re-plugging, or re-installing `opencv2`, or your camera driver, or make sure your camera is compatible with opencv2."
+        )
+
+    return camera_ids
+
+
+def benchmark_cameras(cameras, out_dir=None, save_images=False, num_warmup_frames=4):
+    if out_dir:
+        out_dir = Path(out_dir)
+        out_dir.mkdir(parents=True, exist_ok=True)
+
+    for _ in range(num_warmup_frames):
+        for camera in cameras:
+            try:
+                camera.capture_image()
+                time.sleep(0.01)
+            except OSError as e:
+                print(e)
+
+    while True:
+        now = time.time()
+        for camera in cameras:
+            color_image = camera.capture_image("bgr" if save_images else "rgb")
+
+            if save_images:
+                image_path = out_dir / f"camera_{camera.camera_index:02}.png"
+                print(f"Write to {image_path}")
+                save_color_image(color_image, image_path, write_shape=True)
+
+        dt_s = time.time() - now
+        dt_ms = dt_s * 1000
+        freq = 1 / dt_s
+        print(f"Latency (ms): {dt_ms:.2f}\tFrequency: {freq:.2f}")
+
+        if save_images:
+            break
+        if cv2.waitKey(1) & 0xFF == ord("q"):
+            break
+
+
+@dataclass
+class OpenCVCameraConfig:
+    """
+    Example of tested options for Intel Real Sense D405:
+
+    ```python
+    OpenCVCameraConfig(30, 640, 480)
+    OpenCVCameraConfig(60, 640, 480)
+    OpenCVCameraConfig(90, 640, 480)
+    OpenCVCameraConfig(30, 1280, 720)
+    ```
+    """
+
+    fps: int | None = None
+    width: int | None = None
+    height: int | None = None
+    color: str = "rgb"
+
+
+class OpenCVCamera:
+    # TODO(rcadene): improve dosctring
+    """
+    https://docs.opencv.org/4.x/d0/da7/videoio_overview.html
+    https://docs.opencv.org/4.x/d4/d15/group__videoio__flags__base.html#ga023786be1ee68a9105bf2e48c700294d
+
+    Example of uage:
+
+    ```python
+    camera = OpenCVCamera(2)
+    camera.connect()
+    color_image = camera.read()
+    ```
+    """
+
+    def __init__(self, camera_index: int, config: OpenCVCameraConfig | None = None, **kwargs):
+        if config is None:
+            config = OpenCVCameraConfig()
```

Comment:
Initialize with default `fps`, `height` and `width` maybe ?

Reply:
Default arguments are different with respect to camera in OpenCV so I kept the same logic.